src/tribler-core/tribler_core/modules/metadata_store/community/remote_query_community.py

```diff
@@ -240,6 +271,9 @@ async def on_remote_select_response(self, peer, response_payload):
         processing_results = await self.mds.process_compressed_mdblob_threaded(response_payload.raw_blob)
         self.logger.info(f"Response result: {processing_results}")
 
+        if isinstance(request, EvaSelectRequest):
+            request.processing_results.set_result(processing_results)
```

Comment:
Before calling `future.set_results`, I recommend checking if it's done first. If the exception gets set in the `on_timeout` before calling `set_results` it will throw an exception.